quantus/helpers/perturb_func.py

```diff
@@ -276,3 +280,67 @@ def translation_y_direction(
 def no_perturbation(arr: np.array, **kwargs) -> np.array:
     """Apply no perturbation to input."""
     return arr
+
+
+def noisy_linear_imputation(
+    arr: np.array,
+    indices: Union[int, Sequence[int], Tuple[np.array]],
+    **kwargs,
+) -> np.array:
+    """
+    Based on https://github.com/tleemann/road_evaluation.
+    Calculates noisy linear imputation for the given array and a list of indices indicating which elements are not
+    included in the mask.
+    """
+    offset_weight = [
+        ((1, 1), 1 / 12),
+        ((0, 1), 1 / 6),
+        ((-1, 1), 1 / 12),
+        ((1, -1), 1 / 12),
+        ((0, -1), 1 / 6),
+        ((-1, -1), 1 / 12),
+        ((1, 0), 1 / 6),
+        ((-1, 0), 1 / 6),
+    ]
+    noise = kwargs.get("noise", 0.01)
+    arr_flat = arr.reshape((arr.shape[0], -1))
+
+    mask = np.ones(arr_flat.shape[1])
+    mask[indices] = 0
+
+    ind_to_var_ids = np.zeros(arr_flat.shape[1], dtype=int)
+    ind_to_var_ids[indices] = np.arange(len(indices))
+
+    # Equation system left-hand side
+    a = lil_matrix((len(indices), len(indices)))
+    # Equation system right-hand side
+    b = np.zeros((len(indices), arr.shape[0]))
+    sum_neighbors = np.ones(len(indices))  # Sum of weights assigned
```

Comment:
Place comment above. Tiny thing but please end your comments with a dot so to comply with the style of the rest of the library